cpp/log/cert_checker.cc

```diff
@@ -359,4 +362,104 @@ StatusOr<bool> CertChecker::IsTrusted(const Cert& cert,
   return false;
 }
 
+util::StatusOr<bool> CertChecker::IsCmsSignedByCert(CMS_ContentInfo* const cms,
+                                                    const Cert& cert) const {
+  if (!cert.IsLoaded()) {
+    LOG(ERROR) << "Can't check cert signer as it's not loaded";
+    return Status(util::error::FAILED_PRECONDITION, "Cert not loaded");
+  }
+
+  // This stack must not be freed as it points into the CMS structure
+  STACK_OF(CMS_SignerInfo)* const signers(CMS_get0_SignerInfos(cms));
+
+  if (signers) {
+    for (int s = 0; s < sk_CMS_SignerInfo_num(signers); ++s) {
+      CMS_SignerInfo* const signer = sk_CMS_SignerInfo_value(signers, s);
+
+      if (CMS_SignerInfo_cert_cmp(signer, cert.x509_) == 0) {
+        return true;
+      }
+    }
+  }
+
+  return false;
+}
+
+
+util::Status CertChecker::UnpackCmsDerBio(BIO* cms_bio_in,
```

Comment:
Done for input bio. Output can be nullptr if you want to verify but not unpack. It's documented in .h file.